src/librustc/middle/diagnostic_items.rs

```diff
@@ -0,0 +1,123 @@
+//! Detecting diagnostic items.
+//!
+//! Diagnostic items are items that represent concepts intrinsic to the language
```

Comment:
This doc comment is wrong, diagnostic items are not intrinsic to the language and neither are the examples below valid.